### ISSUE TITLE
Can't render ViewComponent with rich_text_area

### ISSUE THREAD

**Issue by helphop** [github/view_component#445]
### Steps to reproduce

I have the following View Component:

```
<%= form_with(model: [post, element]) do |form| %>
  <%= form.rich_text_area :content %>
  <div class="mt-3 d-flex justify-content-between">
    <div>
      <%= form.submit "Save", class: 'btn btn-primary' %>
      <a href="#" class="cancel btn btn-secondary">Cancel</a>
    </div>
      <%= link_to 'Delete', post_element_path(post, element), method: :delete, class: 'btn btn-danger', data: { confirm: 'Are you sure?'} %>
  </div>
<% end %>
```
### Expected behavior
This should render the Trix Action Text Editor

### Actual behavior
ActionView::Template::Error (undefined method `rich_text_area_tag' for #<ParagraphFormComponent:0x00007f9734219b48>)

### System configuration
**Rails version**:
Rails 6.0.3.2
**Ruby version**:
ruby 2.6.5p114 (2019-10-01 revision 67812) [x86_64-darwin19]
**Gem version**:
Using view_component 2.18.1

**Comment by helphop** [comment]
Can someone please look into this I need to be able to use rich_text_area in my view components.

**Comment by helphop** [comment]
I was able to figure out how to get this working. I hope this helps others with a similar problem.

In your  View Component Class add the following:

```
require 'action_text'
class ParagraphFormComponent < ViewComponent::Base
  include ActionText::Engine.helpers

   def main_app
     Rails.application.class.routes.url_helpers
  end

```
 In config environment development (or other environment file):

`Rails.application.routes.default_url_options[:host] = "localhost"`

It might be helpful to require the action_text in the ViewComponent::Base so it's available to all components as well as the ActionText::Engine.helpers for future releases.

**Comment by joelhawksley** [comment]
@helphop sorry for the delay in getting to this issue, we've been pretty swamped here at work 😅 

Would you be willing to open a PR with a failing test for us to look at? I'd also be happy to pair with you on fixing this issue. Send me an email: joelhawksley@github.com

**Comment by helphop** [comment]
I have forked the application and added the following code to the view_component/base.rb class:

require "action_view"
require "active_support/configurable"
require "view_component/collection"
require "view_component/compile_cache"
require "view_component/previewable"
require "view_component/slotable"
require "action_text/engine"

```
module ViewComponent
  class Base < ActionView::Base
    include ActiveSupport::Configurable
    include ViewComponent::Previewable
    include ActionText::Engine.helpers
    # For CSRF authenticity tokens in forms
    delegate :form_authenticity_token, :protect_against_forgery?, :config, to: :helpers

    class_attribute :content_areas
    self.content_areas = [] # class_attribute:default doesn't work until Rails 5.2

    # Hash of registered Slots
    class_attribute :slots
    self.slots = {}

    #required for rich_text_area
    def main_app
      Rails.application.class.routes.url_helpers
    end
```

I then have loaded this local gem into my project and it works.  The problem is that the tests for the gem do not pass and I have no idea what is causing the error.  Help appreciated as this is the first time I'm helping on a gem.

**Comment by rickychilcott** [comment]
Just highlighting that this is still an issue.

Would it be helpful to have a failing case to resolve the issue?

**Comment by joelhawksley** [comment]
@rickychilcott yes, it would be!

**Comment by jaredplanter** [comment]
@rickychilcott @joelhawksley Has there been any movement on this and is there anything I can do to help it along?

My issue is identical to the original post here. Thanks!

**Comment by boardfish** [comment]
@jaredplanter There doesn't appear to have been from what I can see. Providing us with a failing test case would be a good place to start in order to get things moving.

**Comment by rickychilcott** [comment]
@jaredplanter -- I spent an hour trying to make up a failing test case and I struck out. The best I can tell, the test case will need to add all of the ActionText migrations and a mock model.

If someone would want to pair with me to create a failing test case that works, I'm open to it (my email is ricky@rakefire.io).


Do note, that I have had the following in production for myself for two months without issues.

```ruby
# frozen_string_literal: true

require "action_text/engine"

class ApplicationComponent < ViewComponent::Base
  include ActionText::Engine.helpers

  private

  def main_app
    @main_app ||= Rails.application.class.routes.url_helpers
  end
end
```

**Comment by jaredplanter** [comment]
@boardfish @rickychilcott I did some more digging through the issues and found this: https://github.com/github/view_component/issues/351#issuecomment-631796883

Following that suggestion, I delegated `rich_text_area` to `helpers` and that gave actiontext access to the helpers it needed. I was able to remove the fix from this thread entirely and not have to do any additional configuration. Now actiontext and view_component are playing nicely together.

I hope this helps others!
